### PR TITLE
Hyperchains btc integration

### PR DESCRIPTION
**Pull request by erlmachinedev**

### CODE REVIEW

**Comment by zjn6868** [comment]
成功了没有

**Comment by gorbak25** [comment]
> 成功了没有

It's working, this PR is closely related to #3506

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM aeternity/builder:1804 as builder
 
 # Add required files to download and compile only the dependencies
-ADD rebar.config rebar.lock Makefile rebar3 rebar.config.script /app/
+ADD rebar.config rebar.lock Makefile rebar3 rebar.config.script apps/aehyperchains/src/contracts/SimpleElection.aes /app/
```
Comment (gorbak25) [inline Dockerfile:4]: Wasn't this blacklisted in .dockerignore?

File: apps/aehyperchains/src/aehc_consensus_hyperchains.erl [hunk 2]
```diff
@@ -421,10 +421,14 @@ state_pre_transform_key_node(KeyNode, _PrevNode, PrevKeyNode, Trees1) ->
...
-                             case protocol_staking_contract_call(Trees1, TxEnv, "protocol_enable()") of
-                                 {ok, Trees2, {tuple, {}}} -> Trees2;
-                                 Err1 -> aec_block_insertion:abort_state_transition({activation_failed, Err1})
-                             end
+                             Res =
+                                 case protocol_staking_contract_call(Trees1, TxEnv, "protocol_enable()") of
+                                     {ok, Trees2, {tuple, {}}} -> Trees2;
+                                     Err1 -> aec_block_insertion:abort_state_transition({activation_failed, Err1})
+                                 end,
+                             %% NOTE Setup of parent chain pointer
```
Comment (gorbak25) [inline apps/aehyperchains/src/aehc_consensus_hyperchains.erl:429]: ???

File: apps/aehyperchains/src/aehc_parent_db.erl [hunk 3]
```diff
@@ -76,43 +77,69 @@ get_commitment(CommitmentHash) ->
...
        [#hc_db_commitment_header{value = DBCommitmentHeader}]
-           = mnesia:read(hc_db_commitment_header, CommitmentHash),
+           = get_commitment_header(CommitmentHash),
        CommitmentHeader = aehc_commitment_header:from_db(DBCommitmentHeader),
        case aehc_commitment_header:hc_pogf_hash(CommitmentHeader) of
             NoFraud ->
                 aehc_commitment:new(CommitmentHeader, no_pogf);
             PoGFHash ->
-                [#hc_db_pogf{value = DBPoGF}] = mnesia:read(hc_db_pogf, PoGFHash),
+                [#hc_db_pogf{value = DBPoGF}] = get_pogf(PoGFHash),
```
Comment (gorbak25) [inline apps/aehyperchains/src/aehc_parent_db.erl:86]: PogF will be moved offchain

**Pull request closed without merging**